tests/deadline_test.cpp

```diff
@@ -0,0 +1,95 @@
+/* Copyright © 2001-2015, Canal TP and/or its affiliates. All rights reserved.
+  
+This file is part of Navitia,
+    the software to build cool stuff with public transport.
+ 
+Hope you'll enjoy and contribute to this project,
+    powered by Canal TP (www.canaltp.fr).
+Help us simplify mobility and open public transport:
+    a non ending quest to the responsive locomotion way of traveling!
+  
+LICENCE: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+   
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+   
+You should have received a copy of the GNU Affero General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+  
+Stay tuned using
+twitter @navitia 
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "utils/deadline.h"
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE deadline_test
+#include <boost/test/unit_test.hpp>
+
+static auto past = boost::posix_time::from_iso_string("19840101T000000");
+static auto future = boost::posix_time::from_iso_string("30190101T000000");
+static auto before_sometime = boost::posix_time::from_iso_string("20190313T154217,9999");
+static auto sometime = boost::posix_time::from_iso_string("20190313T154218");
+static auto after_sometime = boost::posix_time::from_iso_string("20190313T154218,0001");
+
+BOOST_AUTO_TEST_CASE(simple_deadline) {
+    auto deadline = navitia::Deadline();
+
+    // an unitialised deadline never expire
+    BOOST_CHECK_EQUAL(deadline.expired(past), false);
+    BOOST_CHECK_EQUAL(deadline.expired(future), false);
+
```

Comment:
```suggestion
BOOST_REQUIRE_NO_THROW(deadline.check());

```

Reply:
To test the real case if 'enable_deadline==false' in source/kraken/kraken_zmq.h (https://github.com/CanalTP/navitia/pull/2710/files#diff-4481db5c3fd4efec2bfad0f126265b68R129)